### PR TITLE
Fix on DS fields defined with start/end position

### PR DESCRIPTION
**Pull request by f-lombardo**
The end position is no more interpretated ad field precision if there exists a start position definition like in

`     D DSTEMP          DS            64
     D VAIDOJ                  1     10A
     D VAATV0                 11     11A`

### CODE REVIEW

**Review by benetti-smeup — COMMENTED** [review]
If precision of positional defined fields was wrong, we need to check unit tests about PACKEVEN, because packeven can only be used with positional defined DS.

**Review by benetti-smeup — COMMENTED** [review]
The PACKEVEN keyword indicates that the packed field or array has an even number of digits.
The keyword is only valid for packed program-described data-structure subfields defined using FROM/TO positions.
if the PACKEVEN keyword is specified, the numberOfDigits is 2(N-1).
If the PACKEVEN keyword is not specified, the numberOfDigits is 2N - 1.

**Review by benetti-smeup — DISMISSED** [review]
It seems that MUTE12_02 fails.
---
~/git-projects/jariko/rpgJavaInterpreter-core/src/test/resources/data/ds/MUTE12_02.rpgle - Total annotation: 157, executed: 0, failed: 0, exceptions: 0, syntax errors: 10
Error(type=LEXICAL, message=token recognition error at: '['                                    COMP(EQ)       ', position=Position(start=Line 1, Column 27, end=Line 1, Column 27))
Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
Error(type=LEXICAL, message=token recognition error at: '['                    '               COMP(EQ)       ', position=Position(start=Line 1, Column 27, end=Line 1, Column 27))
Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 12, end=Line 1, Column 17))
Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 23, end=Line 1, Column 27))
Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 27, end=Line 1, Column 28))
Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 28, end=Line 1, Column 30))
Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 30, end=Line 1, Column 31))

**Comment by f-lombardo** [comment]
> PACKEVEN

Ouch, I didn't know this keyword! So, do you think that we should create a specific test for it? Would you like to write it?

**Comment by f-lombardo** [comment]
> 
> ## It seems that MUTE12_02 fails.
> 
> ~/git-projects/jariko/rpgJavaInterpreter-core/src/test/resources/data/ds/MUTE12_02.rpgle - Total annotation: 157, executed: 0, failed: 0, exceptions: 0, syntax errors: 10
> Error(type=LEXICAL, message=token recognition error at: '[' COMP(EQ) ', position=Position(start=Line 1, Column 27, end=Line 1, Column 27))
> Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
> Error(type=LEXICAL, message=token recognition error at: '[' ' COMP(EQ) ', position=Position(start=Line 1, Column 27, end=Line 1, Column 27))
> Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
> Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
> Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 12, end=Line 1, Column 17))
> Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 23, end=Line 1, Column 27))
> Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 27, end=Line 1, Column 28))
> Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 28, end=Line 1, Column 30))
> Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 30, end=Line 1, Column 31))

I deleted the wrong MUTE test

**Comment by benetti-smeup** [comment]
> > ## It seems that MUTE12_02 fails.
> > ~/git-projects/jariko/rpgJavaInterpreter-core/src/test/resources/data/ds/MUTE12_02.rpgle - Total annotation: 157, executed: 0, failed: 0, exceptions: 0, syntax errors: 10
> > Error(type=LEXICAL, message=token recognition error at: '[' COMP(EQ) ', position=Position(start=Line 1, Column 27, end=Line 1, Column 27))
> > Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
> > Error(type=LEXICAL, message=token recognition error at: '[' ' COMP(EQ) ', position=Position(start=Line 1, Column 27, end=Line 1, Column 27))
> > Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
> > Error(type=SYNTACTIC, message=Recognition exception: null, position=Position(start=Line 1, Column 8, end=Line 1, Column 8))
> > Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 12, end=Line 1, Column 17))
> > Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 23, end=Line 1, Column 27))
> > Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 27, end=Line 1, Column 28))
> > Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 28, end=Line 1, Column 30))
> > Error(type=SYNTACTIC, message=Error node found, position=Position(start=Line 1, Column 30, end=Line 1, Column 31))
> 
> I deleted the wrong MUTE test

Ok

**Comment by benetti-smeup** [comment]
> > PACKEVEN
> 
> Ouch, I didn't know this keyword! So, do you think that we should create a specific test for it? Would you like to write it?

MUTE12_02 contains a specific test for it.
The version in rpgJavaInterpreter-core/src/test/resources/overlay/ seems good.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt [hunk 1]
```diff
@@ -400,12 +400,16 @@ fun RpgParser.Parm_fixedContext.toTypeInfo(): TypeInfo {
 
 internal fun RpgParser.Parm_fixedContext.calculateExplicitElementType(arraySizeDeclared: Int?): Type? {
     val rpgCodeType = DATA_TYPE()?.text?.trim() ?: RpgType.ZONED.rpgType
-    val precision = if (TO_POSITION().text.isNotBlank()) TO_POSITION().text.trim().toInt() else null
     val decimalPositions = if (DECIMAL_POSITIONS().text.isNotBlank()) with(DECIMAL_POSITIONS().text.trim()) { if (isEmpty()) 0 else toInt() } else null
     val isPackEven = keyword().any { it.keyword_packeven() != null }
     val isVarying = keyword().any { it.keyword_varying() != null }
     val startPosition = this.explicitStartOffset()
     val endPosition = this.explicitEndOffset()
+    val precision = if (startPosition != null && endPosition != null) {
```
Comment (ftomassetti) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt:408]: Si potrebbe anche pensare di spostarlo in un extension method separato

**Review by ftomassetti — APPROVED** [review]
Qui mi sembra tutto chiaro

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt (hunk 1 above)
Comment (f-lombardo) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt:408]: Scusa, ma non penso di aver capito il tuo suggerimento. Di che tipo vorresti fare l'estensione?

Comment (ftomassetti) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt:408]: Qualcosa del tipo:
```
fun RpgParser.Parm_fixedContext.precision() =  if (startPosition != null && endPosition != null) {endPosition - startPosition.toInt()
    } else {
        if (TO_POSITION().text.isNotBlank()) TO_POSITION().text.trim().toInt() else null
    }
```

O anche:

```
val RpgParser.Parm_fixedContext.precision
get() =  if (startPosition != null && endPosition != null) {endPosition - startPosition.toInt()
    } else {
        if (TO_POSITION().text.isNotBlank()) TO_POSITION().text.trim().toInt() else null
    }
```

Comment (f-lombardo) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt:408]: startPosition ed endPosition non sono attributi di RpgParser.Parm_fixedContex, ma variabili calcolate in calculateExplicitElementType ed usate in diversi punti della funzione. Ci devo pensare

**Pull request merged**